great_expectations/datasource/data_connector/util.py

```diff
@@ -290,6 +288,49 @@ def get_filesystem_one_level_directory_glob_path_list(
     return path_list
 
 
+def list_azure_keys(
+    azure: BlobServiceClient,
+    query_options: dict,
+    recursive: bool = False,
+) -> List[str]:
+    """
+    Utilizes the Azure Blob Storage connection object to retrieve blob names based on user-provided criteria.
+
+    For InferredAssetAzureDataConnector, we take container and name_starts_with and search for files using RegEx at and below the level
+    specified by those parameters. However, for ConfiguredAssetAzureDataConnector, we take container and name_starts_with and
+    search for files using RegEx only at the level specified by that bucket and prefix.
+
+    This restriction for the ConfiguredAssetAzureDataConnector is needed, because paths on Azure are comprised not only the leaf file name
+    but the full path that includes both the prefix and the file name.  Otherwise, in the situations where multiple data assets
+    share levels of a directory tree, matching files to data assets will not be possible, due to the path ambiguity.
+
```

Comment:
Thank you so much for this thorough description

Reply:
Stole this one from the S3 docstring 😬